Huggingface private model support via API tokens (FARMReader)

ArzelaAscoIi opened the pull request:
Private models currently need to be downloaded in advance and integrated via a local import. To avoid this extra step, it should be possible to pass the `use_auth_token` parameter to the `FARMReader`. The reader uses the 'token' either created by the `transformer-cli` or passed directly as a string to authenticate at [Huggingface](huggingface.co). 

**Proposed changes**:
- pass parameter `use_auth_token` from `FARMReader` to model loading by adding `kwargs` as parameter

**Status (please check what you already did)**:
- [x] First draft (up for discussions & feedback)
- [x] Final code
- [ ] Added tests
- [x] Updated documentation

Timoeller reviewed (commented):
In the past we have seen that passing kwargs around can cause a lot of confusion for users, because it 
1. can silently swallow miss-spelled parameters or 
2. it can set parameters in an unwanted way. Do you know that all kwargs params in Haystacks  LanguageModel.load method are compatible with downstream usage, e.g. ```AutoConfig.from_pretrained(model_name_or_path, **kwargs)```

If you know what you are passing around and can make it default to None without breaking previous code, I would make the parameter explicit. 
Lets also not overcomplicate things since we might be able to comletely switch to HF LanguageModels in the future, e.g. see the TODO in line 77.
WDYT?

tholor reviewed (commented):
Nice! Any particular reason why you didn't make it an explicit parameter on the highest level (in the FARMReader itself)? I think it will be a useful param for our community but users might not be aware of the existence if the param is not exposed + documented explicitly. 

On a side note: This is a great start, but we will eventually need this also for other model classes (e.g. [DensePassageRetriever](https://github.com/deepset-ai/haystack/blob/c0892717a061eef0d9a0a185b0ec1b53f0b467d3/haystack/nodes/retriever/dense.py#L32) or [EmbeddingRetriever](https://github.com/deepset-ai/haystack/blob/c0892717a061eef0d9a0a185b0ec1b53f0b467d3/haystack/nodes/retriever/dense.py#L933)). We can tackle this in separate PRs though.

ArzelaAscoIi commented:
Thanks @tholor  and @Timoeller ! I added `use_auth_token` in FARMReader and added this parameter explicitly to 
to the functions used.

I tested: 
- [x] passing tokens via `transformer-cli login` and setting `use_auth_token=True`
- [x] passing tokens via Access Tokens from huggingface  (`use_auth_token=<your-token>`)

tholor commented on haystack/nodes/reader/farm.py:
```diff
@@ -58,6 +58,7 @@ def __init__(
         proxies=None,
         local_files_only=False,
         force_download=False,
+        use_auth_token: Optional[Union[str,bool]] = None,
```
Please add the new param to the docstring below explaining for our end users what kind of token this is and ideally linking some HF documentation on how to obtain it.
